### PR TITLE
storage CSI e2e: Move csi driver cleanup functions into a common one

### LINKED ISSUE #99054: CSI e2e: refactor cleanup functions into a common one
Tracking bug for https://github.com/kubernetes/kubernetes/pull/98555#discussion_r572335376

/sig storage
/important-longterm
/assign

### PR DESCRIPTION
**Pull request by verult**
#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #99054

#### Special notes for your reviewer:
Manually tested two CSI mock driver test cases

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/sig storage
/priority important-longterm
/triage accepted
/assign @gnufied @msau42

### CODE REVIEW

File: test/e2e/storage/drivers/csi.go [hunk 1]
```diff
@@ -707,3 +652,35 @@ func tryFunc(f func()) error {
 	f()
 	return err
 }
+
+func generateCleanupFunc(
```
Comment (msau42) [inline test/e2e/storage/drivers/csi.go:656]: Nit: Maybe call this "generateDriverCleanupFunc" to be more specific.

**Review by msau42 — COMMENTED** [review]
/approve
Just a naming nit

File: test/e2e/storage/drivers/csi.go (hunk 1 above)
Comment (verult) [reply test/e2e/storage/drivers/csi.go]: done

**Pull request merged**